src/codegen/mod.rs

```diff
@@ -859,7 +892,8 @@ fn gen_exec(
                 })
                 .collect::<Vec<_>>();
             let mut all_param_decls = param_decls;
-            all_param_decls.extend(nat_param_decls);
+            // Why did we need that?
```

Comment:
This adds the nat args as normals arguments. If we need to use shared memory the nats need to be marked as const therefore cannot be used as normal arguments